src/NexusMods.App.UI/Pages/Settings/SettingsPageViewModel.cs

```diff
@@ -0,0 +1,76 @@
+using System.Collections.ObjectModel;
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Linq;
+using DynamicData;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.Settings;
+using NexusMods.App.UI.Controls.Settings.SettingEntries;
+using NexusMods.App.UI.Windows;
+using NexusMods.App.UI.WorkspaceSystem;
+using ReactiveUI;
+using ReactiveUI.Fody.Helpers;
+
+namespace NexusMods.App.UI.Pages.Settings;
+
+[UsedImplicitly]
+public class SettingsPageViewModel : APageViewModel<ISettingsPageViewModel>, ISettingsPageViewModel
+{
+    public ReactiveCommand<Unit, Unit> SaveCommand { get; }
+    public ReactiveCommand<Unit, Unit> CancelCommand { get; }
+    public ReactiveCommand<Unit, Unit> CloseCommand { get; }
+    public ReadOnlyObservableCollection<ISettingEntryViewModel> SettingEntries { get; }
+
+    [Reactive]
+    public bool HasAnyValueChanged { get; private set; }
+
+    public SettingsPageViewModel(ISettingsManager settingsManager, IWindowManager windowManager) : base(windowManager)
+    {
+        CancelCommand = ReactiveCommand.Create(() => { });
+        CloseCommand = ReactiveCommand.Create(() => { });
+
+        var descriptors = settingsManager.GetAllUIProperties();
+        var entryViewModels = descriptors.Select(CreateEntryViewModel).ToArray();
+
+        // ReSharper disable ArrangeObjectCreationWhenTypeNotEvident
+        SettingEntries = new(new(entryViewModels));
+        // ReSharper restore ArrangeObjectCreationWhenTypeNotEvident
+
+        SaveCommand = ReactiveCommand.Create(() =>
+        {
+            var changedEntries = SettingEntries
+                .Where(vm => vm.InteractionControlViewModel.ValueContainer.HasChanged)
+                .ToArray();
+
+            if (changedEntries.Length == 0) return;
+            foreach (var viewModel in changedEntries)
+            {
+                viewModel.InteractionControlViewModel.ValueContainer.Update(settingsManager);
```

Comment:
Save button should also update the `HasChanged` in the value container, so that pressing it makes it disable again. Alternatively, it should close the page.